apps/core/src/utils/transaction/getTransactionAction.ts

```diff
@@ -0,0 +1,27 @@
+// Copyright (c) Mysten Labs, Inc.
+// Modifications Copyright (c) 2024 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+import { IotaTransactionBlockResponse } from '@iota/iota-sdk/client';
+import { TransactionAction } from '../../interfaces';
+import { STAKING_REQUEST_EVENT, UNSTAKING_REQUEST_EVENT } from '../../constants';
+
+export const getTransactionAction = (
+    transaction: IotaTransactionBlockResponse,
+    currentAddress?: string,
+) => {
+    const stakeTypeTransaction = transaction?.events?.find(
+        ({ type }) => type === STAKING_REQUEST_EVENT,
+    );
+    const unstakeTypeTransaction = transaction?.events?.find(
+        ({ type }) => type === UNSTAKING_REQUEST_EVENT,
+    );
+    if (stakeTypeTransaction) {
```

Comment:
We could include here timelocked staking and unstaking, we have events for these too, see events in https://explorer.rebased.iota.org/txblock/8QshWrqFhiLYxsFbcx3rcT9pNJm1oh1RJ4jJh5eL57Vk?network=testnet & https://explorer.rebased.iota.org/txblock/5qJeMgWwvar5ZE58EUWBMmLsenBqmpCJeRsuYTsDvetf?network=testnet